internal/e2e/lds_test.go

```diff
@@ -70,6 +71,58 @@ func TestNonTLSListener(t *testing.T) {
 	}, fetchLDS(t, cc))
 }
 
+func TestTLSListener(t *testing.T) {
+	rh, cc, done := setup(t)
+	defer done()
+
+	// s1 is a tls secret
+	s1 := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "secret",
+			Namespace: "default",
+		},
+		Data: map[string][]byte{
+			v1.TLSCertKey:       []byte("certificate"),
+			v1.TLSPrivateKeyKey: []byte("key"),
+		},
+	}
+
+	// i1 is a tls ingress
+	i1 := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "simple",
+			Namespace: "default",
+		},
+		Spec: v1beta1.IngressSpec{
+			Backend: backend("backend", intstr.FromInt(80)),
+			TLS: []v1beta1.IngressTLS{{
+				Hosts:      []string{"kuard.example.com"},
+				SecretName: "secret",
+			}},
+		},
+	}
+
+	// add secret, ingest and assert the existence of ingress_http and ingres_https
+	rh.OnAdd(s1)
+	rh.OnAdd(i1)
+	assertEqual(t, &v2.DiscoveryResponse{
+		VersionInfo: "0",
+		Resources: []*types.Any{
+			any(t, listener("ingress_http", "0.0.0.0", 8080,
+				filterchain(false, httpfilter("ingress_http")),
+			)),
+			any(t, listener("ingress_https", "0.0.0.0", 8443,
+				filterchaintls(
+					[]string{"kuard.example.com"},
+					"certificate", "key",
+					false, httpfilter("ingress_https")),
+			)),
+		},
+		TypeUrl: cgrpc.ListenerType,
+		Nonce:   "0",
+	}, fetchLDS(t, cc))
```

Comment:
Please add a final test to delete the secret and verify ingress_https is removed
```
rh.OnDelete(s1)
        assertEqual(t, &v2.DiscoveryResponse{
                VersionInfo: "0",
                Resources:   []*types.Any{},
                TypeUrl:     cgrpc.ListenerType,
                Nonce:       "0",
        }, fetchLDS(t, cc))
```